dev/io.openliberty.reporting.internal/src/io/openliberty/reporting/internal/DataCollector.java

```diff
@@ -0,0 +1,165 @@
+import com.ibm.ws.kernel.productinfo.DuplicateProductInfoException;
+import com.ibm.ws.kernel.productinfo.ProductInfo;
+import com.ibm.ws.kernel.productinfo.ProductInfoParseException;
+import com.ibm.ws.kernel.productinfo.ProductInfoReplaceException;
+
+import io.openliberty.reporting.internal.utils.HashUtils;
+
+/**
+ * <p>
+ * Collects the required data from the kernel to pass to the cloud service to
+ * check if there is any CVE's that could have impact.
+ * </p>
+ */
+public class DataCollector {
+
+	/**
+	 * 
+	 * @return data Map<String, String>
+	 */
+	public Map<String, String> getData() {
+		Map<String, String> data = new HashMap<String, String>();
+
+		data.put("id", uniqueID);
+		data.put("productEdition", productEdition);
+		data.put("productVersion", productVersion);
+		data.put("features", String.join(",", installedFeatures));
+		data.put("javaVendor", javaVendor);
+		data.put("javaVersion", javaRuntimeInfo);
+		data.put("iFixes", String.join(",", iFixSet));
+		data.put("osArch", osArch);
+		data.put("os", os);
+
+		return data;
+	}
+
+	private final String uniqueID;
+
+	private String javaVendor = "";
+	private final String javaRuntimeInfo;
+
+	private final Set<String> installedFeatures = new HashSet<String>();
+
+	private final String productVersion;
+	private final String productEdition;
+
+	private final Set<String> iFixSet = new HashSet<String>();
+
+	private final String os;
+
+	private final String osArch;
+
+	/**
+	 * <p>
+	 * Collects all the required data.
+	 * </p>
+	 * 
+	 * @param featureProvisioner
+	 * @param FixManager
+	 * @param serverInfo
+	 * @throws IOException
+	 * @throws DataCollectorException
+	 */
+	public DataCollector(FeatureProvisioner featureProvisioner, FixManager FixManager, ServerInfoMBean serverInfo)
+			throws IOException, DataCollectorException {
+
+		Map<String, ? extends ProductInfo> allProductInfo;
+		try {
+			allProductInfo = ProductInfo.getAllProductInfo();
+
+			uniqueID = HashUtils.hashString(serverInfo.getInstallDirectory()); // placeholder.
+
+			javaRuntimeInfo = serverInfo.getJavaRuntimeVersion();
+
+			installedFeatures.addAll(getPublicFeatures(featureProvisioner));
+
+			String javaVM = System.getProperty("java.vendor").toLowerCase();
+
+			if (javaVM == null) {
+				javaVM = System.getProperty("java.vm.name", "unknown").toLowerCase();
+			}
+
+			javaVendor = javaVM;
+
+			iFixSet.addAll(FixManager.getIFixes());
+
+			os = System.getProperty("os.name");
+
+			osArch = System.getProperty("os.arch");
+
+			// the key is the productId
+			if (allProductInfo.containsKey("com.ibm.websphere.appserver")) {
+				productVersion = allProductInfo.get("com.ibm.websphere.appserver").getVersion();
+				productEdition = allProductInfo.get("com.ibm.websphere.appserver").getEdition();
+			} else {
+				productVersion = allProductInfo.get("io.openliberty").getVersion();
+				productEdition = allProductInfo.get("io.openliberty").getEdition();
```

Comment:
This assumes that allProductInfo will have `"io.openliberty"` in it.  If it doesn't this will be a NullPointerException.  Is it guaranteed to have io.openliberty in it?

Reply:
As far as I am aware if the product is not WebSphere Liberty it will have to be Open Liberty.